zk_toolbox/crates/zk_supervisor/src/commands/test/revert.rs

```diff
@@ -1,54 +1,67 @@
-use common::{cmd::Cmd, logger, spinner::Spinner};
+use std::path::PathBuf;
+
+use anyhow::Context;
+use common::{cmd::Cmd, config::global_config, logger, spinner::Spinner};
 use config::EcosystemConfig;
+use ethers::utils::hex::ToHex;
 use xshell::{cmd, Shell};
 
-use super::args::revert::RevertArgs;
+use super::{
+    args::revert::RevertArgs,
+    utils::{install_and_build_dependencies, TestWallets, TEST_WALLETS_PATH},
+};
 use crate::messages::{
-    msg_revert_tests_run, MSG_REVERT_TEST_INSTALLING_DEPENDENCIES, MSG_REVERT_TEST_RUN_INFO,
-    MSG_REVERT_TEST_RUN_SUCCESS,
+    msg_revert_tests_run, MSG_CHAIN_NOT_FOUND_ERR, MSG_DESERIALIZE_TEST_WALLETS_ERR,
+    MSG_REVERT_TEST_RUN_INFO, MSG_REVERT_TEST_RUN_SUCCESS,
 };
 
 const REVERT_TESTS_PATH: &str = "core/tests/revert-test";
 
-pub fn run(shell: &Shell, args: RevertArgs) -> anyhow::Result<()> {
+pub async fn run(shell: &Shell, args: RevertArgs) -> anyhow::Result<()> {
     let ecosystem_config = EcosystemConfig::from_file(shell)?;
     shell.change_dir(ecosystem_config.link_to_code.join(REVERT_TESTS_PATH));
 
     logger::info(MSG_REVERT_TEST_RUN_INFO);
-    install_and_build_dependencies(shell, &ecosystem_config)?;
-    run_test(shell, &args, &ecosystem_config)?;
-    logger::outro(MSG_REVERT_TEST_RUN_SUCCESS);
 
-    Ok(())
-}
+    if !args.no_deps {
+        install_and_build_dependencies(shell, &ecosystem_config)?;
+    }
 
-fn install_and_build_dependencies(
-    shell: &Shell,
-    ecosystem_config: &EcosystemConfig,
-) -> anyhow::Result<()> {
-    let _dir_guard = shell.push_dir(&ecosystem_config.link_to_code);
-    let spinner = Spinner::new(MSG_REVERT_TEST_INSTALLING_DEPENDENCIES);
-    Cmd::new(cmd!(shell, "yarn install")).run()?;
-    Cmd::new(cmd!(shell, "yarn utils build")).run()?;
+    run_test(shell, &args, &ecosystem_config).await?;
+    logger::outro(MSG_REVERT_TEST_RUN_SUCCESS);
 
-    spinner.finish();
     Ok(())
 }
 
-fn run_test(
+async fn run_test(
     shell: &Shell,
     args: &RevertArgs,
     ecosystem_config: &EcosystemConfig,
 ) -> anyhow::Result<()> {
     Spinner::new(&msg_revert_tests_run(args.external_node)).freeze();
 
+    let chain_config = ecosystem_config
+        .load_chain(global_config().chain_name.clone())
+        .context(MSG_CHAIN_NOT_FOUND_ERR)?;
+
+    let wallets_path: PathBuf = ecosystem_config.link_to_code.join(TEST_WALLETS_PATH);
+    let wallets: TestWallets = serde_json::from_str(shell.read_file(&wallets_path)?.as_ref())
+        .context(MSG_DESERIALIZE_TEST_WALLETS_ERR)?;
+
+    wallets
+        .init_test_wallet(ecosystem_config, &chain_config)
+        .await?;
+    let private_key = wallets.get_test_wallet(&chain_config)?.private_key.unwrap();
```

Comment:
looks like you have the same code for all tests

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2594/commits/a47effab904422f0a9c0911ba04cd6c4923106dd